### PR TITLE
Optimera RTDM: Added support for ORTB data injection with updated compatibility with "waitForIt" delay

### PR DESCRIPTION
**Pull request by kcandiotti**
## Type of change

- [ ] Bugfix
- [x] Feature
- [ ] New bidder adapter  
- [ ] Updated bidder adapter   
- [x] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [no] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

## Description of change

The Optimera RTDM fetches what we call a "scorefile" once per Prebid initialization. This is a .js file which contains a JSON array of targeting values. Traditionally, these were made ready for ad server (GAM) key value setting and passing. However, we have new interest in passing our data into bidders as part of their bid request. Therefore, we have added support to pass our data into the ORTB2 object. Since our data is slot level and not page level, we are making use of the ORTB2Imp.ext.data object, adding data under the key "optimera" in addition to passing our data into the ad platform (GAM) as we used to. As part of this update, we found that our adapter was not properly communicating with the waitForIt command, resulting in our response coming back too late to make the bid request. Therefore, we updated accordingly and all of our testing shows that the data is fetched in time and is available and being passed into bid adapter bid requests in the proper space. Lint and Unit testing passed.

## Other information

This is our stab at doing this. While it works for our purposes, if there is anything that is not following normal convention please let us know and we will update accordingly.

### CODE REVIEW

File: modules/optimeraRtdProvider.js [hunk 1]
```diff
@@ -30,167 +28,84 @@ import { ajaxBuilder } from '../src/ajax.js';
...
 let _moduleParams = {};
 
-/**
- * Default Optimera Key Name
- * This can default to hb_deal_optimera for publishers
- * who used the previous Optimera Bidder Adapter.
- * @type {string}
- */
+// Default key name used in legacy targeting
+/** @type {string} */
```
Comment (patmmccann) [inline modules/optimeraRtdProvider.js:32]: you're welcome to submit a .ts adapter

File: modules/optimeraRtdProvider.md [hunk 2]
```diff
@@ -1,46 +1,54 @@
...
-| device  | string  | optional | Device type code for mobile, tablet, or desktop. Either mo, tb, de |
-| apiVersion  | string  | optional | Optimera API Versions. Either v0, or v1. ** Note: v1 wll need to be enabled specifically for your account, otherwise use v0.
\ No newline at end of file
+| Parameter Name             | Type    | Scope     | Description |
+|----------------------------|---------|-----------|-------------|
+| `clientID`                 | string  | required  | Optimera Client ID. Contact Optimera to obtain yours. |
+| `optimeraKeyName`          | string  | optional  | GAM key name for Optimera targeting. Defaults to `hb_deal_optimera` for legacy compatibility. |
+| `device`                   | string  | optional  | Device type code. Use `mo` (mobile), `tb` (tablet), or `de` (desktop) or output the common library splitter value here. |
+| `apiVersion`               | string  | optional  | Optimera API version. Allowed values: `v0` or `v1`. **Note:** `v1` must be enabled by Optimera. |
+| `transmitWithBidRequests` | string  | optional  | Set to `'allow'` to inject Optimera data into the ORTB2 object for bid requests. |
```
Comment (patmmccann) [inline modules/optimeraRtdProvider.md]: please note the default if not set

File: modules/optimeraRtdProvider.js [hunk 3]
```diff
@@ -231,34 +178,48 @@ export function setScores(result) {
...
+export function injectOrtbScores(reqBidsConfigObj) {
+  reqBidsConfigObj.adUnits.forEach((adUnit) => {
+    const auCode = adUnit.code;
+    adUnit.ortb2Imp = adUnit.ortb2Imp || {};
+    adUnit.ortb2Imp.ext = adUnit.ortb2Imp.ext || {};
+    adUnit.ortb2Imp.ext.data = adUnit.ortb2Imp.ext.data || {};
+
+    if (auCode && optimeraTargeting[auCode]) {
+      mergeDeep(adUnit.ortb2Imp.ext.data, {
+        optimera: optimeraTargeting[auCode]
```
Comment (patmmccann) [inline modules/optimeraRtdProvider.js:205]: would be helpful to know the structure of  optimeraTargeting[auCode]

**Comment by kcandiotti** [comment]
Regarding the "unit test label" is that something you need us to do or is that an internal label that someone on the prebid side just needs to run the test? Just want to make sure we are doing something if we need to.

**Comment by patmmccann** [comment]
> Regarding the "unit test label" is that something you need us to do or is that an internal label that someone on the prebid side just needs to run the test? Just want to make sure we are doing something if we need to.

please see https://github.com/prebid/Prebid.js/blob/master/CONTRIBUTING.md#writing-tests

you should cover all your code in tests

**Comment by kcandiotti** [comment]
We updated our spec file, all tests are green and believe this covers anything that was newly added. Please let us know if anything else is needed, thank you.

**Comment by kcandiotti** [comment]
Also, we just wanted to flag that the end to end tests are failing because of other adapter code. We don't think there is anything we can do to resolve that but let us know if this is a blocker.

**Comment by kcandiotti** [comment]
@dgirardi - I think you may have ran the tests and skipped "Run tests / Unit tests (all features enabled + coverage)" and "Run tests / Update" which unclogged the pipes but now it gets stuck at two more. Those failing tests seems to be from adapters / code unrelated to our changes / adapter. Could those two be skipped as well?

**Comment by kcandiotti** [comment]
@dgirardi - Thank you for the update to get the tests to pass. I see we are all green but were not part of Prebid's release on Thursday. Is there anything else we need to do to make this ready to ship? Thank you!
